core/domain/state_domain.py

```diff
@@ -2052,6 +2333,15 @@ def validate(self) -> None:
                 'Expected needs_update to be a bool, received %s' %
                 self.needs_update)
 
+        # Validate translations.
+        if isinstance(self.translation, list):
+            for html in self.translation:
+                validate_rte_tags(html)
+                validate_tabs_and_collapsible_rte_tags(html)
+        else:
```

Comment:
Better to explictily check the format of the translation and handle it accordingly rather than have a default case like this.

Reply:
I'm sorry, I didn't fully understand this part. According to me the data that we can receive in translation can be of this type - "<p>Content</p>" or ["<p>Content1</p>", "<p>Content2</p>"]. Do you mean that I should use `isinstance` here?

Reply:
See my comment above regarding use of data_format to determine the structure of translation and then handle the translation accordingly. We shoud avoid the scenario where we "default case" a validation -- this can lead to unexpected errors. It is harder to maintain too e.g. consider a scenario where we add support for a new data format of translations and this if-else block is not updated.

Reply:
Thanks, I understand it now. This part will only run when the type will be `html` but I would still need to check if the incoming data is coming in a list format or just str, right? Does the current sounds good, please let me know if I have to change something.